nestkernel/connection_creator_impl.h

```diff
@@ -110,6 +115,56 @@ ConnectionCreator::connect_to_target_( Iterator from,
   }
 }
 
+template < typename Iterator, int D >
+void
+ConnectionCreator::connect_to_target_poisson_( Iterator from,
+  Iterator to,
+  Node* tgt_ptr,
+  const Position< D >& tgt_pos,
+  size_t tgt_thread,
+  const Layer< D >& source )
+{
+  RngPtr rng = get_vp_specific_rng( tgt_thread );
+
+  // We create a source pos vector here that can be updated with the
+  // source position. This is done to avoid creating and destroying
+  // unnecessarily many vectors.
+  std::vector< double > source_pos( D );
+  const std::vector< double > target_pos = tgt_pos.get_vector();
+
+  // Declare number of connections variable
+  unsigned long num_conns;
+
+  const bool without_kernel = not kernel_.get();
+  for ( Iterator iter = from; iter != to; ++iter )
+  {
+    if ( not allow_autapses_ and ( iter->second == tgt_ptr->get_node_id() ) )
+    {
+      continue;
+    }
+    iter->first.get_vector( source_pos );
+
+    // Sample number of connections that are to be established
+    num_conns = rng->prand( kernel_->value( rng, source_pos, target_pos, source, tgt_ptr ) );
+    if ( without_kernel or num_conns )
```

Comment:
`without_kernel` will have the same value for all iterations of the loop, won't it? Then you could return at the top of the method if it is `false`. For clarity, I would write `num_conns > 0`—although you don't need it since the subsequent loop is not run if `num_conns == 0`.  In the loop, I would stick to C++ conventions and let it run from zero.

Reply:
Addressed in 9af0d30ec25d3d223a5283a4c638e54b25fbaebc.

Regarding ```without_kernel```: I am not sure you can move it outside, at least not without breaking the current logic, c.f [here](https://github.com/nest/nest-simulator/blob/12cff3d370f503130142e575b3e1ea202a92ce49/nestkernel/connection_creator_impl.h#L97): if ```rng->drand() < kernel_->value( rng, source_pos, target_pos, source, tgt_ptr )``` evaluates to ```false``` but ```without_kernel``` to true a connection is established.
This realization also lead to a bug fix of the current code to make it consistent with the cod  [here](https://github.com/nest/nest-simulator/blob/12cff3d370f503130142e575b3e1ea202a92ce49).
But what this case should catch is unclear to me - maybe I don't understand it and thus don't understand your comment.

Reply:
After reconsidering:
The ```without_kernel``` is not needed at this point, nor does it need to be checked at this level.
An error is thrown immediately if the required parameter ```pairwise_avg_num_conns``` is not provided.